engine-nextgen/src/main/kotlin/graphql/nadel/NextgenEngine.kt

```diff
@@ -139,15 +139,18 @@ class NextgenEngine @JvmOverloads constructor(
             executionPlan,
         )
         val transformedQuery = queryTransform.result.single()
-        val result = executeService(service, transformedQuery, executionContext)
-        val transformedResult = resultTransformer.transform(
-            executionContext = executionContext,
-            executionPlan = executionPlan,
-            artificialFields = queryTransform.artificialFields,
-            overallToUnderlyingFields = queryTransform.overallToUnderlyingFields,
-            service = service,
-            result = result,
-        )
+        val result: ServiceExecutionResult = executeService(service, transformedQuery, executionContext)
+        val transformedResult: ServiceExecutionResult = when {
+            topLevelField.name.startsWith("__") -> result
```

Comment:
Can you use something from `graphql-java` to check if this is an introspection field? Like `Introspection.isIntrospectionTypes`? Although that takes a `GraphQLNamedType`...

Reply:
Tbh this kinda sucks. We should really skip the entire process of execution planning etc. for introspection queries as there's really nothing to do at all.

But it's fine for now, let's revisit this in the future.